private-media-framework/draft-ietf-perc-private-media-framework.md

```diff
@@ -548,37 +558,53 @@ re-keyed is outside the scope of this document, but this framework
 does accommodate re-keying during the life of a conference.
 
 When a Key Distributor decides to re-key a conference, it transmits a
-specific message defined in PERC EKT [@!I-D.ietf-perc-srtp-ekt-diet] to
-each of the conference participants.  The endpoint **MUST** create a
+new EKTKey message [@!I-D.ietf-perc-srtp-ekt-diet] to
+each of the conference participants containing the new EKT Key.
+Upon receipt of the new EKT Key, the endpoint **MUST** create a
 new SRTP master key and prepare to send that key inside a Full EKT
-Field using the new EKTKey.  Since it may take some time for all of the
-endpoints in conference to finish re-keying, senders **MUST** delay a
-short period of time before sending media encrypted with the new
-master key, but it **MUST** be prepared to make use of the information
-from a new inbound EKT Key immediately. See Section 2.2.2 of
-[@!I-D.ietf-perc-srtp-ekt-diet].
+Field using the new EKT Key as per Section 4.5 of [@!I-D.ietf-perc-srtp-ekt-diet].
+Since it may take some time for all of the endpoints in conference to finish
+re-keying, senders **MUST** delay a short period of time before sending media
+encrypted with the new master key and before utilizing the new EKT Key
+(continuing using the old EKT Key temporarily), but it **MUST** be prepared to
+process packets containing the new EKT Key immediately. The EKT SPI field can
+be used to differentiate between tags encrypted with the old and new keys.
+
+> Editor's Note: Now much time is a "short period of time"? It would be
+  really nice if the EKTKey message contained this delay. And the KD knows
+  roughly how long it would take to send out messages. For a two-party
+  conference, this might be 100ms, but for a large conference it might be
+  several seconds.
```

Comment:
True that 4.5 in DIET does not explicitly indicate how quickly or not a sender must send packets with the new SRTP master.  But it does implicitly suggest expedience as key management would likely do a rekey because a change that impacts privacy (eg, roster change) such that subsequent traffic can "only be decrypted by endpoints that know the current EKTKey."   Thus, I can understand why Ben & Richard might say what they did.

Personally, I think any implementer focused on user experience will opt for a short delay to increase odds of successful decrypt (and avoid blank/frozen periods) by those that will (eventually) know the current EKTKey, trading off the potential downside of sending a few syllables/video frame that might get sent with the prior master after a rekey event.  However, in retrospect, we should not have made it a "MUST", but perhaps a "RECOMMEND" instead.  It is an implementer decision on how to balance user experience vs privacy/security needs during a rekey event.

Since it is not a requirement for interoperability, and since congestion between a KD and endpoints may not be known by a KD, I don't think we need exact delay communicated in a message from the KD.

How to proceed as authors?   I am ok if we drop the sender's "delay short period" point altogether as Ben suggested or convert this to a recommendation, with some text acknowledging the tradeoff(s).